idea/data-storage/src/gear/gear.service.ts

```diff
@@ -88,71 +95,88 @@ export class GearService {
     this.logger.log(`⚙️ Connected to ${this.api.runtimeChain} with genesis ${this.genesis}`);
     reconnectionsCounter = 0;
     this.unsub = await this.listen();
+    this.indexBlocks();
     changeStatus('gear');
   }
 
   private async listen() {
-    return this.api.derive.chain.subscribeNewBlocks(async (block) => {
-      const blockHash = block.block.hash.toHex();
-      const timestamp = (await this.api.blocks.getBlockTimestamp(block)).toNumber();
-
-      await this.handleBlock(block, timestamp, blockHash);
-      await this.handleExtrinsics(block, timestamp);
-
-      for (const {
-        event: { data, method },
-      } of block.events) {
-        try {
-          const payload = getPayloadByGearEvent(method, data as GenericEventData);
-          if (payload !== null) {
-            await this.handleEvents(method, { ...payload, blockHash, timestamp, genesis: this.genesis });
-          }
-        } catch (error) {
-          console.error(error);
-          this.logger.warn({ method, data: data.toHuman() });
+    this.newBlocks = [];
+    return this.api.rpc.chain.subscribeFinalizedHeads(({ number, hash }) => {
+      this.newBlocks.push({ number: number.toNumber(), hash: hash.toHex() });
+    });
+  }
+
+  private async *blocksGenerator() {
+    while (true) {
+      if (this.newBlocks.length === 0) {
+        await sleep(1000);
+        continue;
+      }
+      yield this.newBlocks.shift();
+    }
+  }
+
+  private async indexBlocks() {
+    for await (const { hash, number } of this.blocksGenerator()) {
```

Comment:
You have the value `number` in the `loop`. Can you rename the value to a more meaningful name to recognize what is the `magic` number value?